### ISSUE TITLE
Shellcheck doesn't understand typeset -x

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#2354]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related
I searched `is:issue SC2034 typeset` and came up with three results, the most relevant of which is probably #958, but I can't tell if it has supposedly been addressed or not with the numerous issue cross-refs within.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bash
foofunc()
{
        echo I am foofunc
}
typeset -fx foofunc
```

#### Here's what shellcheck currently says:
```
Line 8	SC2034: foofunc appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

Nothing. From bash manpage, section typeset:
```
-f     Use function names only.
-x     Mark names for export to subsequent commands via the environment.
```

Because I am exporting them, they can be used elsewhere. Therefore, I contend that SC2034 (appears unused. Verify it or export it.) does not apply.

**Comment by koalaman** [comment]
`typeset` is now treated the same as `declare`, as it should have been all along. Thanks!

**Comment by masalomon** [comment]
Just BTW: This _was_ a previous issue, #1922.  Thanks for the fix.
